stripped/colabs/Regresja_logistyczna_i_SVM_stripped.py

```diff
@@ -0,0 +1,1180 @@
+             crossval_results = Kfolds_crossvalidation(model, folds, folds_labels)
+             results.update(crossval_results)
+             results_sorted = dict({key : results[key] for key in all_res.keys()})
+             all_res = {key: all_res[key] + [results_sorted[key]] for key in all_res}
+
+  cross_results = pd.DataFrame(data=all_res)
+
+  return cross_results
+
+param_grid_linear = { 'penalty': ['l1', 'l2'],
+               'C': [0.01, 0.1, 0.5, 0.8, 1, 5, 10, 50, 100],
+               'loss': ['squared_hinge'],                           #with loss='hinge' convergence issues occurred
+               'class_weight': [None, 'balanced', {1:1.5, 0:1}],
+               'intercept_scal': [0.1, 1, 1.5, 5, 10, 50]}
+
+
+#Splitting into folds
+blocks = block_partition(July_2019_DF, 2.5, min_lat, max_lat, min_lon, max_lon)
+folds_July_2019 = Kfolds(blocks, 6, random_state=3)
+folds_07_2019_labels = [list(fold['pustynia']) for fold in folds_July_2019]
+
+#Data standarization
+folds_July_2019 = [fold.drop(columns=['lon', 'lat', 'Year', 'Month','pustynia','SoilT_40_100cm'])
+                   for fold in folds_July_2019]
+folds_07_2019_stand = [data_standarization(fold) for fold in folds_July_2019]
+
+linear_results1 = GridSearchCV_LinearSVM(folds_07_2019_stand, folds_07_2019_labels, param_grid_linear)
+linear_results.sort_values(by=['accuracy', 'precision'], ascending=False)
+
+#Splitting into folds
+blocks = block_partition(August_2019_DF, 2.5, min_lat, max_lat, min_lon, max_lon)
+folds_August_2019 = Kfolds(blocks, 6, random_state=3)
+folds_08_2019_labels = [list(fold['pustynia']) for fold in folds_August_2019]
+
+#Data standarization
+folds_August_2019 = [fold.drop(columns=['lon', 'lat', 'Year', 'Month','pustynia','SoilT_40_100cm'])
+                   for fold in folds_August_2019]
+folds_08_2019_stand = [data_standarization(fold) for fold in folds_August_2019]
+
+linear_results2 = GridSearchCV_LinearSVM(folds_08_2019_stand, folds_08_2019_labels, param_grid_linear)
+linear_results2.sort_values(by=['accuracy', 'precision'], ascending=False)
+
+#Splitting into folds
+blocks = block_partition(December_2019_DF, 2.5, min_lat, max_lat, min_lon, max_lon)
+folds_December_2019 = Kfolds(blocks, 6, random_state=3)
+folds_12_2019_labels = [list(fold['pustynia']) for fold in folds_December_2019]
+
+#Data standarization
+folds_December_2019 = [fold.drop(columns=['lon', 'lat', 'Year', 'Month','pustynia','SoilT_40_100cm'])
+                   for fold in folds_December_2019]
+folds_12_2019_stand = [data_standarization(fold) for fold in folds_December_2019]
+
+linear_results2 = GridSearchCV_LinearSVM(folds_12_2019_stand, folds_12_2019_labels, param_grid_linear)
+linear_results2.sort_values(by=['accuracy', 'precision'], ascending=False)
+
+def GridSearchCV_SVM(folds: np.ndarray, folds_labels: List[List[int]], param_grid: dict, kernel: str) -> pd.DataFrame:
+
```

Comment:
W zasadzie niemal identyczne funkcja do poprzedniej. Nie da się ich zgeneralizować do jednej na pewno?

Reply:
Można "zmergować" tę funkcję z tą poprzednią: będzie to wymagało w  zasadzie przekopiowania calego kodu tej funkcji do wnętrza tej poprzedniej (LinearSCV ma zupelnie inne parametry niz SCV). Zrobić to?

Reply:
w takich przypadkach zrób parametr o nazwie hyperparams: Dict[Any,Any] i wypakuj go do modelu.
```
self.model = sklearn_model(*hyperparams)
```

Reply:
Przerobiłam funkcję. Nie mam pewności, czy zgodnie z oczekiwaniami...